docs/scripting-element-object.rst

```diff
@@ -0,0 +1,566 @@
+
+element:text
+------------
+
+Fetch a text information from the element
+
+**Signature:** ``text = element:text()``
+
+**Returns:** ``text`` is a text content
+of the element.
+
+**Async:** no.
+
+It tries to return the trimmed value of the following JavaScript ``Node`` properties:
+
+* textContent
+* innerText
+* value
+
+If all of them are empty an empty string is returned.
+
+
+.. _splash-element-info:
+
+element:info
+------------
+
+Get useful information about the element.
+
+**Signature:** ``info = element:info()``
+
+**Returns:** ``info`` is a table with element info.
+
+**Async:** no.
+
+Info is a table with the following fields:
+
+* nodeName - node name in a lower case (e.g. *h1*)
+* attributes - table with attributes names and its values
+* tag - html string representation of the element
+* html - inner html of the element
+* text - inner text of the element
+* x - x coordinate of the element
+* y - y coordinate of the element
+* width - width of the element
+* height - height of the element
+* visible - flag representing if the element is visible
+
+
+.. _splash-element-field-value:
+
+element:field_value
+-------------------
+
+Get value of the field element (input, select).
+
+**Signature:** ``ok, info = element:field_value()``
+
+**Returns:** ``ok, value`` pair. If ``ok`` is nil then error happened during the function call;
+``value`` provides an information about error type; otherwise ``value`` is a value of the
+element.
+
+**Async:** no.
+
+The value can be a
+* string - for text/radio input and select and other element types
+* array of strings - for multi select
+* boolean - for checkbox input
+
+
+.. _splash-element-form-values:
+
+element:form_values
+-------------------
+
+Return a table with form values if the element type is *form*
+
+**Signature:** ``ok, info = element:form_values()``
+
+**Returns:** ``ok, values`` pair. If ``ok`` is nil then error happened during the function call
+or node type is not *form*; ``values`` provides an information about error type; otherwise
+``values`` is a table of values.
+
+**Async:** no.
+
+
+.. _splash-element-fill:
+
+element:fill
+------------
+
+Fill the form with the provided values
+
+**Signature:** ``ok, reason = element:fill(values)``
+
+**Parameters:**
+
+* values - table with input names as keys and values as input values
+
+**Returns:** ``ok, reason`` pair. If ``ok`` is nil then error happened during the
+function call; ``reason`` provides an information about error type.
+
+**Async:** no.
+
+In order to fill your form your inputs must have ``name`` property and this method will
+select those input using that property.
+
+Example of filling the following form:
+
+.. code-block:: html
+
+    <form id="login">
+        <input type="text" name="username" />
+        <input type="password" name="password" />
+    </form>
+
+.. code-block:: lua
+
+    function main(splash)
+        assert(splash:select('.login'):fill({ username="admin", password="pass" }))
+    end
```

Comment:
Is it possible to get form values using elem:form_values(), change a couple of field values and call splash:fill() - are values compatible?